shell/platform/tizen/channels/key_event_channel.cc

```diff
@@ -28,195 +30,220 @@ constexpr char kKeyDown[] = "keydown";
+    {0x0000009e, 269025070},  // launchInternetBrowser
+    {0x000000a3, 269025049},  // launchMail
+    {0x000000a4, 269025072},  // browserFavorites
+    {0x000000a6, 269025062},  // browserBack
+    {0x000000a7, 269025063},  // browserForward
+    {0x000000a9, 269025068},  // eject
+    {0x000000ab, 269025047},  // mediaTrackNext
+    {0x000000ad, 269025046},  // mediaTrackPrevious
+    {0x000000ae, 269025045},  // mediaStop
+    {0x000000af, 269025052},  // mediaRecord
+    {0x000000b0, 269025086},  // mediaRewind
+    {0x000000b1, 269025134},  // launchPhone
+    {0x000000b4, 269025048},  // browserHome
+    {0x000000b5, 269025065},  // browserRefresh
+    {0x000000bd, 269025128},  // newKey
+    {0x000000be, 65382},      // redo
+    {0x000000bf, 65482},      // f13
+    {0x000000c0, 65483},      // f14
+    {0x000000c1, 65484},      // f15
+    {0x000000c2, 65485},      // f16
+    {0x000000c3, 65486},      // f17
+    {0x000000c4, 65487},      // f18
+    {0x000000c5, 65488},      // f19
+    {0x000000c6, 65489},      // f20
+    {0x000000c7, 65490},      // f21
+    {0x000000c8, 65491},      // f22
+    {0x000000c9, 65492},      // f23
+    {0x000000ca, 65493},      // f24
+    {0x000000d1, 269025073},  // mediaPause
+    {0x000000d6, 269025110},  // close
+    {0x000000d7, 269025044},  // mediaPlay
+    {0x000000d8, 269025175},  // mediaFastForward
+    {0x000000da, 65377},      // print
+    {0x000000e1, 269025051},  // browserSearch
+    {0x000000e8, 269025027},  // brightnessDown
+    {0x000000e9, 269025026},  // brightnessUp
+    {0x000000ed, 269025030},  // kbdIllumDown
+    {0x000000ee, 269025029},  // kbdIllumUp
+    {0x000000ef, 269025147},  // mailSend
+    {0x000000f0, 269025138},  // mailReply
+    {0x000000f1, 269025168},  // mailForward
+    {0x000000f2, 269025143},  // save
+    {0x00000190, 269025170},  // launchAudioBrowser
+    {0x00000195, 269025056},  // launchCalendar
+    {0x000001aa, 269025163},  // zoomIn
+    {0x000001ab, 269025164},  // zoomOut
+    {0x000001b8, 269025148},  // spellCheck
+    {0x000001b9, 269025121},  // logOff
+    {0x0000024d, 269025069},  // launchScreenSaver
+};
+
+// Mapping from Ecore modifiers to GTK modifiers.
+//
+// The values are originally defined in:
 // - efl/Ecore_Input.h
 // - flutter/raw_keyboard_linux.dart (GtkKeyHelper)
 const std::map<int, int> kModifierMap = {
     {0x0001, 1 << 0},   // SHIFT (modifierShift)
     {0x0002, 1 << 2},   // CTRL (modifierControl)
     {0x0004, 1 << 3},   // ALT (modifierMod1)
-    {0x0008, 1 << 28},  // WIN (modifierMeta)
+    {0x0008, 1 << 26},  // WIN (modifierMeta)
     {0x0010, 0},        // SCROLL (undefined)
     {0x0020, 1 << 4},   // NUM (modifierMod2)
     {0x0040, 1 << 1},   // CAPS (modifierCapsLock)
 };
 
+uint32_t Utf8ToUtf32CodePoint(const char* utf8) {
+  std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> converter;
+  for (auto wchar : converter.from_bytes(utf8)) {
+    return std::uint_least32_t(wchar);
+  }
```

Comment:
Could you explain why you are using a loop and what std::uint_least32_t is used for?
Maybe it's because I'm not familiar with these but I didn't understand :cry:

Reply:
> why you are using a loop

As you know `const char*` is a string type so it may contain multiple characters (although we expect it to contain only a single character in this case). The loop is used to obtain the first character from the string using `wstring_convert`'s built-in iterator.

> what std::uint_least32_t is used for

I just copied that part from Internet 😛. `uint_least32_t` is a typedef of `char32_­t`, and it's also equivalent to `uint32_t` in our configuration. It should be safe to just remove the cast.

Reply:
Oh, I see, thank you for answer! :)